lambdas/thumbnail/src/t4_lambda_thumbnail/__init__.py

```diff
@@ -322,6 +325,44 @@ def handle_pptx(*, src: bytes, page: int, size: int, count_pages: bool):
     return info, data
 
 
+def generate_thumbnail(arr, size):
+    # Send to Image object for thumbnail generation and saving to bytes
+    img = Image.fromarray(arr)
+
+    # The mode I;16 has limited resamplers for scaling, and throws an error.
+    # Rather than use a non-default poor-quality resampler, convert to a better-handled mode.
+    if img.mode == 'I;16' and not _TEST_FALLBACK:
+        img = Image.fromarray((arr // 256).astype('uint8'))
```

Comment:
If it's hard to find/generate an image that actually hits the fallback case then you can patch (or wrap) Image.fromarray to return`fromarray(arr)` and that effectively skips the above line.

Reply:
Another thing you could do (maybe even cleaner) is make a sub function that takes the image mode as a string and just patch that value on a specific test run to skip the above line.

Reply:
> patch (or wrap) Image.fromarray

true....

> make a sub function

I'll do that.

Reply:
Alright.  My only remaining reservation is that after handling the PIL `ValueError("image has wrong mode")` and failing, I still think we should raise our own error.  But overall, I'm ok with what's here now.

Reply:
Feel free to do error handling as you prefer: raise our own.